src/components/new-safe/create/steps/Step2/index.tsx

```diff
@@ -64,6 +64,7 @@ const CreateSafeStep2 = ({
     // Set threshold if it's greater than the number of owners
     setValue(CreateSafeStep2Fields.threshold, Math.min(threshold, ownerFields.length - 1))
     remove(index)
+    trigger()
```

Comment:
```suggestion
    trigger(CreateSafeStep2Fields.owners)
```
Not a big difference but its enough to trigger validation for owners only